### ISSUE TITLE
Validating depending if the field exists in the DOM

### ISSUE THREAD

**Issue by idesignpixels** [jaredpalmer/formik#683]
I'm building a quite a complex form which conditionally shows fields inside a field array item based on sibling field values.
My question is in redux-form they only validate fields that exist in the DOM but whilst using formik and yup it fails validation even if the field doesn't exist, how to I only validate fields that have been conditionally added?
The conditional fields are select fields that only want to be rendered if options are available to them otherwise don't render.

The rough structure is:
form
--field
--field-array
---tabs
----tab-1
-----field
----^conditional field
----^conditional field
----tab-2
-----conditional fields
--field

I need to check two things, that tab-1's fields are valid before going to tab-2 and that all fields for a field array item are valid before adding another.

**Comment by latviancoder** [comment]
Formik has no idea about your dom, so when you hide field from your dom your also have to manually remove the field value from values object (using setValues or setFieldValue).

**Comment by idesignpixels** [comment]
Sure, but what if the field has no value and is required?
Typically you build the schema like all fields exist but when on doesn't it still tries to validate it as having no value.

**Comment by jaredpalmer** [comment]
When I finish #684, i will add `currentFields` (or a better name) to the FormikProps. This would include the mounted fields

**Comment by idesignpixels** [comment]
@jaredpalmer That could be quite handy, My current fix involves having values that aren't field values to validate alongside for example.

-card
--cardColor
---value
---options // (16 values dynamically updates)

Then use yup when method to set the cardColor>value to be required when cardColor>options are > 0

Currently I'm having to update the options with props into initialValues however this resets the values to the props each time so I have to make a reference to the form values (this.latestValues = values;) then in initialValues do {...this.latestValues, cards: props.cards}

I'm hoping #522 may help as I could replace these helper values with contexts however it appears yup has no concept of an array index when validating so I can't tailor validation based on index position.

I also saw an issue somewhere talking about an initialValuesChange handler which may also help.

**Comment by Andreyco** [comment]
You can solve this right away by moving validation from form level to [field level ](https://github.com/jaredpalmer/formik#validate-value-any--undefined--string--promiseany)

**Comment by davemooreuws** [comment]
@jaredpalmer Are there plans to merge `currentFields` or what ever you would like to call it in as per your post on Jun 13?

**Comment by idoo** [comment]
@jaredpalmer yep, `currentFields` will solve my pain, any chance to get it soon?

**Comment by mouhsnimohamed** [comment]
I'm facing the same issue! was this problem resolved?

**Comment by idoo** [comment]
@mouhsnimohamed still exist. Fee; free to add PR ;)

**Comment by milkysingh** [comment]
Has anyone been able to solve this? I have a similar issue in which I am showing a couple of form fields depending on the checkbox selected/unselected and I want validation on the fields when they appear and no validation when fields dissappears.

**Comment by Ciaran0** [comment]
Coming across the same problem here, current fields would be very useful.

**Comment by hegelstad** [comment]
This would be very powerful in tandem with the validationSchema prop. Is this being worked on?

**Comment by harshane** [comment]
I am still facing this issue, was anyone able to get around this?

**Comment by MEsteves22** [comment]
Did someone solve this issue? I am unfortunately having the same problem :/

**Comment by hrushilok** [comment]
I am facing same issue. Has anyone have any workaround for this?

**Comment by Senelith** [comment]
I need this so badly in my project. Is there a workaround to achieve this ?

**Comment by rusyurchenko** [comment]
Same here, the Formik shouldn't trigger validation on conditional fields unless they were mounted to the DOM and should remove conditional field entry when a field gets unmounted.

**Comment by kostiantyn-solianyk** [comment]
@jaredpalmer  so, any updates on this? It's 2020, but still exist.

**Comment by juanrouges** [comment]
I end the up here for the same reason. I have a field that its required but in some cases its hidden. Is this something formik handles now? or is there a work around?

Just trying a stab in the dark since I see a bunch of other comments with no answer.

**Comment by amilaDulanjana** [comment]
I am also having this issue. Any worker around for this ?

**Comment by kostiantyn-solianyk** [comment]
I think for now we can only optionally change validation schema depending on visible/hidden fields

**Comment by k3nda** [comment]
Hi there, any update please? This is really crucial thing, which should be implemented in Formik.

**Comment by Hajzenberg** [comment]
This would really help.

**Comment by mg-83** [comment]
Have this issue too

**Comment by vipul261** [comment]
@idesignpixels and others who are facing this issue, what helped me solve this conditional validation is modifying my `Yup` validation schema and adding dependency logic in it. In my case I wanted to show some of the fields only when a parent field has a certain value. For that I wrapped my original field validation schema as show below:
```
dependentField: Yup.mixed().when("<PARENT_FIELD_ID>", {
            is: value => value && value.length > 0, // CHECK IF YOU NEED TO VALIDATE THE CURRENT FIELD
            then: <PASS_ORIGINAL_SCHEMA>
            otherwise: Yup.mixed() // No validation
        })
```
Refer this: https://github.com/jquense/yup/issues/736

I hope this helps

**Comment by emrahaydemir** [comment]
Hello my friend who has this problem. A very simple solution to this problem is `$exist`. Here is an example for you: 

``` js
repeatPassword: yup.string().when("$exist", {
    is: (exist) => exist,
    then: yup.string().required("Password repeat is required!),
    otherwise: yup.string(),
  }),

**Comment by Dhmsjoshi** [comment]
@jaredpalmer :I have a same issue. I have a switch button which changes fields on switch from login mode to signup mode. in signup mode, there is username and in login mode it is hidden. this switch works on a state (isLoginMode) . but when I use conditional statement in yup like when isLoginMode is : true then validate . bt it is notworking. it still validated entire form whether it its present on DOM or not. How to solve it?
